src/Symfony/Component/Console/Command/LockableTrait.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Command;
+
+use Symfony\Component\Console\Exception\LogicException;
+use Symfony\Component\Console\Exception\RuntimeException;
+use Symfony\Component\Filesystem\LockHandler;
+
+/**
+ * Basic lock feature for commands.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+trait LockableTrait
+{
+    protected $lock;
+
+    /**
+     * Locks a command.
+     *
+     * @return bool
+     */
+    protected function lock($name = null, $blocking = false)
+    {
+        if (!class_exists('Symfony\Component\Filesystem\LockHandler')) {
+            throw new RuntimeException('To enable the locking feature you must install the symfony/filesystem component');
+        }
+
+        if ($this->lock !== null) {
+            throw new LogicException('A lock is already in place');
+        }
+
+        $this->lock = new LockHandler($name ?: $this->getName());
+
+        if (!$this->lock->lock($blocking)) {
+            $this->release();
```

Comment:
`$this->lock = null` is enough, right ?.

 And it's not related, but:

* I would rename the property `lock` to `lockHandler` (it is not a lock, but something that lock/release a lock)
* I would add a dot at the end of every exception message